src/Microsoft.ML.Data/DataLoadSave/CompositeDataLoader.cs

```diff
@@ -110,5 +113,25 @@ private static VersionInfo GetVersionInfo()
                 loaderSignature: LoaderSignature,
                 loaderAssemblyName: typeof(CompositeDataLoader<,>).Assembly.FullName);
         }
+
+        #region IDisposable Support
+        private bool _disposed;
+
+        private void Dispose(bool disposing)
```

Comment:
📝 This is the legacy `IDisposable` pattern. The new way to implement this is moving this implementation code directly into `IDisposable.Dispose()`.

Reply:
Weren't there two cases to be handled before? The one that calls Dispose(true) from from IDisposable.Dispose() and the other that calls Dispose(false) from the finalizer? (Or someplace else...I am not quite sure). How would that be handled?

---
In reply to: [396132625](https://github.com/dotnet/machinelearning/pull/4939#discussion_r396132625) [](ancestors = 396132625)

Reply:
>  the other that calls Dispose(false) from the finalizer

Finalizers are not allowed under the new dispose pattern.